tools/run_tests/jobset.py

```diff
@@ -229,15 +226,12 @@ def state(self, update_cache):
       self._tempfile.seek(0)
       stdout = self._tempfile.read()
       filtered_stdout = _filter_stdout(stdout)
```

Comment:
I think the logic for escaping (_filter_stdout) and truncating the stdout  should be part of the report generation. Also the same logic 
```
filtered_stdout = _filter_stdout(stdout)
filtered_stdout = filtered_stdout[-128:]
```

is used multiple times in run_tests.py and run_interop_tests.py, and this PR sound like a good opportunity to do some cleanup.

Also, the the TODO is out of date and we can delete it. And we should probably return last 8KB of stdout, 128 bytes is not enough to display any meaningful troubleshooting info (like stack trace).